image/scanner/konflux/Dockerfile

```diff
@@ -0,0 +1,62 @@
+ARG BASE_REGISTRY=registry.access.redhat.com
+ARG BASE_IMAGE=ubi8-minimal
+ARG BASE_TAG=latest
+
+FROM brew.registry.redhat.io/rh-osbs/openshift-golang-builder:rhel_8_1.20 as builder
+
+ENV CGO_ENABLED=1
+ENV GOFLAGS=""
+ENV CI=1
+
+COPY . /src
+WORKDIR /src
+
+RUN unzip -j blob-repo2cpe.zip -d image/scanner/dump/repo2cpe && \
+    unzip -j blob-k8s-definitions.zip -d image/scanner/dump/k8s_definitions && \
+    unzip -j blob-nvd-definitions.zip -d image/scanner/dump/nvd_definitions
+
+RUN echo "version: $(git describe --tags --abbrev=10 --dirty --long)"; git status; go build -trimpath -ldflags="-X github.com/stackrox/scanner/pkg/version.Version=$(git describe --tags --abbrev=10 --dirty --long)" -o image/scanner/bin/scanner ./cmd/clair
+
+# Replace genesis manifests file in the source code with the one generated at
+# the point when the dump was taken.  This is to avoid discrepancy between other
+# files of the dump and the manifest.
+COPY ./blob-genesis_manifests.json image/scanner/dump/genesis_manifests.json
+
+FROM ${BASE_REGISTRY}/${BASE_IMAGE}:${BASE_TAG}
+
+LABEL name="scanner" \
+      vendor="StackRox" \
+      maintainer="support@stackrox.com" \
+      summary="Image scanner for the StackRox Kubernetes Security Platform" \
+      description="This image supports image scanning in the StackRox Kubernetes Security Platform."
+
+ENV NVD_DEFINITIONS_DIR="/nvd_definitions"
+ENV K8S_DEFINITIONS_DIR="/k8s_definitions"
+ENV REPO_TO_CPE_DIR="/repo2cpe"
+
+COPY --from=builder /src/image/scanner/scripts /
+COPY --from=builder /src/image/scanner/bin/scanner ./
+COPY --chown=65534:65534 --from=builder "/src/image/scanner/dump${NVD_DEFINITIONS_DIR}/" ".${NVD_DEFINITIONS_DIR}/"
+COPY --chown=65534:65534 --from=builder "/src/image/scanner/dump${K8S_DEFINITIONS_DIR}/" ".${K8S_DEFINITIONS_DIR}/"
+COPY --chown=65534:65534 --from=builder "/src/image/scanner/dump${REPO_TO_CPE_DIR}/" ".${REPO_TO_CPE_DIR}/"
+COPY --chown=65534:65534 --from=builder /src/image/scanner/dump/genesis_manifests.json ./
+
+RUN microdnf upgrade --nobest && \
+    microdnf install xz && \
```

Comment:
I don't know why the downstream dockerfile installs `findutils` as I can't spot any calls to that in `*.sh` code. Maybe some code existed but was removed. Maybe this was added or copied without much consideration. Maybe this was copied for manual debuggability (I, for instance, like it much more when container where I `exec` has `find` as opposed when it doesn't).

For the last point, I want to establish a baseline of utils to be present in every our image so that they are useful when poking them manually. But that's certainly outside of scope of our RHTAP onboarding.

If you don't see a reason to have `findutils`, let's proceed without them and this comment is no-op.

It's fine to not try to install `ca-certificates` because they are already present in recent ubi-minimal versions.

Reply:
You added `findutils` to the downstream dockerfile in commit 93cf4b3c7b356 (in the rhacs-midstream repo), but it looks like Ross fixed it in scanner by removing the calls to `find` in 28fbde2c5304640da6dd1ff08371ff102b605513.  So it should be safe to omit it from the dockerfile now.

Reply:
Thanks for investigating this. It has been a long time and I forgot.